user/views.py

```diff
@@ -14,3 +25,115 @@ class UserRetrieveDeleteView(generics.RetrieveDestroyAPIView):
     lookup_field = 'pk'
     queryset = User.get_all_users()
     serializer_class = UserSerializer
+
+
+class SendOTPView(views.APIView):
+    permission_classes = (AllowAny,)
+
+    def post(self, request: Request):
+        email = request.data.get('email')
+        client_id = request.data.get('client_id')
+
+        client = Application.objects.get(client_id=client_id)
+
+        if not is_valid_email(email):
+            return Response({
+                'message': 'Invalid email address.'
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        if User.objects.filter(email=email).exists():
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        otp: SignUpOTP = SignUpOTP.get_existing_otp_or_none(email, client)
+
+        if not otp:
+            otp: SignUpOTP = SignUpOTP.create_otp_for_email(email, client)
+
+        if otp.is_blocked():
+            return Response({
+                'message': 'This email has been temporarily blocked. please try after some time.'
+            }, status.HTTP_403_FORBIDDEN)
+        else:
+            otp.update_resends()
+            otp.save()
+
+        verification_message = get_verification_message_with_code(otp.one_time_code)
+        send_mail.delay('HS: Please Verify your Email', verification_message, [otp.email])
+
+        return Response({
+            'message': 'OTP Sent Successfully.'
+        }, status=status.HTTP_200_OK)
+
+
+class VerifyOTPView(views.APIView):
+    permission_classes = (AllowAny,)
+
+    def post(self, request: Request):
+        email = request.data.get('email')
+        client_id = request.data.get('client_id')
+        otp_string = request.data.get('otp')
+
+        if is_valid_email(email) and User.objects.filter(email=email).exists():
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        client = Application.objects.get(client_id=client_id)
+
+        otp: SignUpOTP = SignUpOTP.get_existing_otp_or_none(email, client)
+
+        if not otp:
+            return Response({
+                'message': 'Bad request'
```

Comment:
'Bad request' will be a very common string to use. You can create a `globals.constants` module for such strings.